test/test_nn.py

```diff
@@ -248,21 +248,27 @@ def _backward(self, module, input, output, grad_output):
             return None
         return input.grad.data
 
-    def _forward_criterion(self, criterion, input, target):
+    def _forward_criterion(self, criterion, input, target, reduce=True):
         if isinstance(input, tuple):
             args = input + (target,)
             output = criterion(*args)
         else:
             output = criterion(input, target)
-        return output.data[0]
+        if reduce:
+            return output.data[0]
+        else:
+            return output.data
 
-    def _backward_criterion(self, criterion, input, target):
+    def _backward_criterion(self, criterion, input, target, gradOutput=None):
         input_tuple = input if isinstance(input, tuple) else (input,)
         for i in input_tuple:
             if i.grad is not None:
                 i.grad.data.zero_()
         args = input_tuple + (target,)
-        criterion(*args).backward()
+        if gradOutput is None:
+            criterion(*args).backward()
+        else:
+            criterion(*args).backward(gradOutput)
```

Comment:
I think you can just always give `gradOutput` to `backward`. It will be ignored if it's `None`

Reply:
Also nit: we use `camel_case` on python for variables. So maybe `grad_output`?